crates/database/src/jit_orders.rs

```diff
@@ -0,0 +1,218 @@
+use {
+    crate::{
+        orders,
+        orders::{BuyTokenDestination, OrderKind, SellTokenSource, SigningScheme},
+        Address,
+        AppId,
+        OrderUid,
+    },
+    sqlx::{
+        types::{
+            chrono::{DateTime, Utc},
+            BigDecimal,
+        },
+        PgConnection,
+    },
+};
+
+const JIT_ORDERS_SELECT: &str = r#"
+o.uid, o.owner, o.creation_timestamp, o.sell_token, o.buy_token, o.sell_amount, o.buy_amount,
+o.valid_to, o.app_data, o.fee_amount, o.kind, o.signature,
+o.receiver, o.signing_scheme, o.sell_token_balance, o.buy_token_balance,
+(SELECT COALESCE(SUM(t.buy_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_buy,
+(SELECT COALESCE(SUM(t.sell_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_sell,
+(SELECT COALESCE(SUM(t.fee_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_fee,
+COALESCE((SELECT SUM(surplus_fee) FROM order_execution oe WHERE oe.order_uid = o.uid), 0) as executed_surplus_fee
+"#;
+
+pub async fn single_full_jit_order(
+    ex: &mut PgConnection,
+    uid: &OrderUid,
+) -> Result<Option<FullJitOrder>, sqlx::Error> {
+    #[rustfmt::skip]
+        const QUERY: &str = const_format::concatcp!(
+"SELECT ", JIT_ORDERS_SELECT,
+" FROM jit_orders o",
+" WHERE o.uid = $1 ",
+        );
+    sqlx::query_as(QUERY).bind(uid).fetch_optional(ex).await
+}
+
+#[derive(Debug, Clone, Default, PartialEq, sqlx::FromRow)]
+pub struct FullJitOrder {
+    pub uid: OrderUid,
+    pub owner: Address,
+    pub creation_timestamp: DateTime<Utc>,
+    pub sell_token: Address,
+    pub buy_token: Address,
+    pub sell_amount: BigDecimal,
+    pub buy_amount: BigDecimal,
+    pub valid_to: i64,
+    pub app_data: AppId,
+    pub fee_amount: BigDecimal,
+    pub kind: OrderKind,
+    pub signature: Vec<u8>,
+    pub sum_sell: BigDecimal,
+    pub sum_buy: BigDecimal,
+    pub sum_fee: BigDecimal,
+    pub receiver: Address,
+    pub signing_scheme: SigningScheme,
+    pub sell_token_balance: SellTokenSource,
+    pub buy_token_balance: BuyTokenDestination,
+    pub executed_surplus_fee: BigDecimal,
+}
+
+#[derive(Debug, Clone, Default, PartialEq, sqlx::FromRow)]
+pub struct JitOrder {
```

Comment:
What's the difference between full and non full jit order? This should be documented.

Reply:
documented